project/src/main/java/com/google/sps/data/UserFeedback.java

```diff
@@ -0,0 +1,89 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.util.Optional;
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+
+/**
+ * Represents the feedback that users enter about the recommendations they were given.
+ */
+@AutoValue
+public abstract class UserFeedback {
+
+  /** @return a list of IDs of the places that were recommended to the user. */
+  public abstract ImmutableList<String> placesRecommendedToUser();
+
+  /** @return the ID of the place the user chose, if any. */
+  public abstract Optional<String> placeUserChose();
+
+  /**
+  * @return true if the user requested to try again and get more recommendations (implying they
+  * weren't satisfied with the recommenrations he recieved).
+  */
+  public abstract boolean userTriedAgain();
+
+
+  /** @return a builder that enables to build a new UserFeedback object. */
+  public static Builder builder() {
+    return new AutoValue_UserFeedback.Builder();
+  }
+
+  /** A builder class for creating a UserFeedback objects. */
+  @AutoValue.Builder
+  public abstract static class Builder {
+    /**
+     * @param placesRecommendedToUser a list of IDs of the places that were recommended to the user.
+     * @return a UserFeedback builder that enables to continue building.
+     */
+    public abstract Builder setPlacesRecommendedToUser(ImmutableList<String>
+        placesRecommendedToUser);
+
+    /**
+     * @param chosenPlaceId the ID of the place the user chose, if any.
+     * @return a UserFeedback builder that enables to continue building.
+     */
+    public abstract Builder setPlaceUserChose(String placeUserChose);
+
+    /**
+     * @param triedAgain true if the user requested to try again and get more recommendations
+     * @return a UserFeedback builder that enables to continue building
+     */
+    public abstract Builder setUserTriedAgain(boolean userTriedAgain);
+
+    /**
+     * Builds the UserFeedback object according to the data that was set so far.
+     *
+     * @return the object that was built
+     */
+    abstract UserFeedback autoBuild();
+
+    /**
+     * Concludes the building of a new UserFeedback instance.
+     *
+     * @return the new instance.
+     * @throws IllegalArgumentException if the user feedback isn't consistent.
+     */
+    public UserFeedback build() throws IllegalArgumentException {
+      UserFeedback feedback = autoBuild();
+      ValidationUtils.validateChosenPlaceInReccomendedPlaces(feedback.placeUserChose(),
+          feedback.placesRecommendedToUser());
+      ValidationUtils.validateUserTriedAgainOnlyIfdidntchoose(feedback.placeUserChose(),
```

Comment:
This can actually easiliy happen, our UI doesn't restrict selecting a place and then pressing try again. If it really bothers us I think we should do something about it in the UI and not just get here errors. Options I thought about:
* somehow end the User's session after he picked a place (maybe add some "hope you enjoy" message and exit the results screen).
* When the user selects a place and then pressess "try again" inform him he can't try again while a place is selected

Another option, that I think is not so bad is to just treat this situation as the user did select that place, even if he pressed try again after wards. 

What do you think?

Reply:
I think you are absolutely right about pointing this out. I'm actually working on the UI in a different PR (where I'm working on the JS and HTML). I think currently I'm implementing the last option you said, but this can be changed. WDYT should be the way to choose? I think either the first or the last. 
Either way, I think it's not required for this current PR, I can add a TODO to remind us this must happen

Reply:
The treatment that is required for this PR is if we choose the last option I mentioned (remove the validation).

I thought of another option which I think can be the best user experience but I don't think we have the time for this change:
When the user picks a place and then presses try again, then leave the option he chose and only replace the two other options. 

Since we are limited in time I would go for the last option, which all it takes is removing this validation, but if you are not comftable with this you can leave it as a todo and we discuss it later.

Reply:
Going over your discussion and looking at the implementation, I'm not sure it's indeed correct to validate this. If the UI enables this behavior, than it's not an invalid argument. Think about the user experience here, what would happen if the user picked a place now and then clicked try-again?

I believe this should first be handled in the UI (if this is what you want to enforce), i.e:
If the user picked a place and then clicked "try-again", if possible, treat it as if there are no choices made (this is something your implementation should handle, and shouldn't harm the user experience IMO).

On any case, whichever solution you opt for, I'd add the validation afterwards and not in this PR.

Reply:
I think it doesn't make sense to keep a listing of a place that is both chosen and tried again. I don't want to handle the FE in this PR, so I don't mind deleting it though I think it's a bit redundant to delete and potentially rewrite.
done.